### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by samiran-bs**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,134 @@
+# Logs
```
Comment (samiran-bs) [inline .gitignore:1]: remove the file above .DS_Store.
This file is not needed.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,293 @@
...
+  `git clone https://github.com/browserstack/browserstack-examples-cypress`
+- Install the following dependencies
+  - cypress >= 6.6.0
+  - typescript >= 4.1.2
+  - webpack >= 4.41.2
+  - mocha >= 8.3.2
+  - mochawesome >= 6.2.2
+  - docker-compose >= 0.23.6
+  - browserstack-cypress-cli >= 1.8.0
+
```
Comment (samiran-bs) [inline README.md]: you can rephrase that these all dependencies will get installed using the npm install command on the root  directory of the repository.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,293 @@
...
+​
+Run your tests on the following infrastructures:
+​
+- [On-premise/self-hosted](#on-premise-self-hosted)
+- [Docker](#docker)
+- [BrowserStack](#browserstack)
+​
+---
+​
+# On Premise / Self Hosted
```
Comment (samiran-bs) [inline README.md:71]: Not reviewing the rest of the README.md as the new template will cover most things.

File: bstack-local-parallel.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        ]
+      }
+    ],
+    "run_settings": {
+      "cypress_config_file": "./cypress.json",
+      "project_name": "BrowserStack",
+      "build_name": "browserstack-examples-cypress",
+      "exclude": [],
+      "cypress_version":"6",
+      "parallels": "10",
```
Comment (samiran-bs) [inline bstack-local-parallel.json]: keep default to 5

File: bstack-local-parallel.json [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      "npm_dependencies": {
+        "cypress": "6.6.0",
```
Comment (samiran-bs) [inline bstack-local-parallel.json]: 6.1

File: bstack-local-single.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        ]
+      }
+    ],
+    "run_settings": {
+      "cypress_config_file": "./cypress.json",
+      "project_name": "BrowserStack",
+      "build_name": "browserstack-examples-cypress",
+      "exclude": [],
+      "cypress_version":"6",
+      "parallels": "10",
```
Comment (samiran-bs) [inline bstack-local-single.json]: keep default to 5

File: bstack-local-single.json [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+      "npm_dependencies": {
+        "cypress": "6.6.0",
```
Comment (samiran-bs) [inline bstack-local-single.json]: 6.1

File: bstack-parallel.json [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+      ]
+    }
+  ],
+  "run_settings": {
+    "cypress_config_file": "./cypress.json",
+    "project_name": "BrowserStack",
+    "build_name": "browserstack-examples-cypress",
+    "exclude": [],
+    "cypress_version":"6",
+    "parallels": "10",
```
Comment (samiran-bs) [inline bstack-parallel.json]: keep default to 5

File: bstack-parallel.json [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    "npm_dependencies": {
+      "cypress": "6.6.0",
```
Comment (samiran-bs) [inline bstack-parallel.json]: 6.1

File: bstack-single.json [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+      ]
+    }
+  ],
+  "run_settings": {
+    "cypress_config_file": "./cypress.json",
+    "project_name": "BrowserStack",
+    "build_name": "browserstack-examples-cypress",
+    "exclude": [],
+    "cypress_version" : "6",
+    "parallels": "10",
```
Comment (samiran-bs) [inline bstack-single.json]: keep default to 5

File: bstack-single.json [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    "npm_dependencies": {
+      "cypress": "6.6.0",
```
Comment (samiran-bs) [inline bstack-single.json]: 6.1

File: cypress/integration/e2e/test8.spec.ts [hunk 12]
```diff
@@ -0,0 +1,89 @@
+import Login from '../../pageObjects/login'
```
Comment (samiran-bs) [inline cypress/integration/e2e/test8.spec.ts]: remove all .DS_Store files above

File: cypress/integration/e2e/test8.spec.ts [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+
+  it('Click on buy button', () => {
+
+    purchase.buyButton().click({force:true});
+
+  })
+
+  it('Update address details', () => {
+
+ 	  address.firstname().should('be.visible').type('first');
```
Comment (samiran-bs) [inline cypress/integration/e2e/test8.spec.ts]: have you used a linter / formatter? keep code aligned by using a formatter

File: cypress/integration/login/test3.spec.ts [hunk 14]
```diff
@@ -0,0 +1,24 @@
+describe('Login Feature', () => {
```
Comment (samiran-bs) [inline cypress/integration/login/test3.spec.ts]: remove .DS_Store file above

File: cypress/integration/login/test3.spec.ts [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+ 
+  var url = Cypress.config().baseUrl + "signin?favourites=true";
```
Comment (samiran-bs) [inline cypress/integration/login/test3.spec.ts]: expected_url OR a better name. I initially thought this is the URL to start the test and later saw that it is being used to assert at the end

File: cypress/integration/login/test4.spec.ts [hunk 16]
```diff
@@ -0,0 +1,46 @@
+
+
+describe('Login as locked_user', function ()  {
+
+
```
Comment (samiran-bs) [inline cypress/integration/login/test4.spec.ts]: use a linter or formatter so that the code is appropriately spaced and looks good to read too :)

File: cypress/integration/offers/test9.spec.ts [hunk 17]
```diff
@@ -0,0 +1,52 @@
+
```
Comment (samiran-bs) [inline cypress/integration/offers/test9.spec.ts]: remove .DS_Store file above

File: cypress/integration/product/test1.spec.ts [hunk 18]
```diff
@@ -0,0 +1,44 @@
+describe('Apply Lowest to Highest Order By', () => {
```
Comment (samiran-bs) [inline cypress/integration/product/test1.spec.ts]: remove .DS_Store file above.

File: cypress/integration/product/test1.spec.ts [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+
+    cy.visit('/');
+
+  })
+
+  it('Order by "lowest to highest"', () => {
+
+    cy.get('.App').should('be.visible');
+  	cy.get('select').select('lowestprice');
+  	cy.wait(2000);
```
Comment (samiran-bs) [inline cypress/integration/product/test1.spec.ts]: [Optional but great to learn] 
I had read Cypress allows intercepting network requests (https://docs.cypress.io/guides/guides/network-requests.html#Testing-Strategies)
when the lowestprice dropdown is selected, there is a network request made to https://bstackdemo.com/api/products. Check network requests tab after selecting lowestprice dropdown.

Now, can you check what is the easiest way to intercept the network request and wait for its completion. and then  there is no requirement for the wait. Sounds interesting? :)

File: cypress/integration/product/test2.spec.ts [hunk 20]
```diff
@@ -0,0 +1,26 @@
+describe('Apply Apple and Samsung Vendor Filter', () => {
+
+  var no_of_items = 16;
```
Comment (samiran-bs) [inline cypress/integration/product/test2.spec.ts]: change var name to "expected_count_of_products"

File: cypress/plugins/cy-ts-preprocessor.js [hunk 21]
```diff
@@ -0,0 +1,27 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
```
Comment (samiran-bs) [inline cypress/plugins/cy-ts-preprocessor.js:1]: .DS_Store above. please remove.

File: cypress/support/commands.ts [hunk 22]
```diff
@@ -0,0 +1,244 @@
...
+      // fillIdPage: typeof fillIdPage;
+      // fillProductPage: typeof fillProductPage;
+      // skipDuplicateCheck: typeof skipDuplicateCheck;
+      google(): Chainable<Window>
+      navigate(pageName: string): void
+    }
+  }
+}
+/**
+ * Goes to google site
```
Comment (samiran-bs) [inline cypress/support/commands.ts]: is this file needed? even if needed, the commented lines can be removed.
surely Goes to Google is not needed.

File: cypress/support/commands.ts [hunk 23]
```diff
@@ -0,0 +1,244 @@
...
+// Convert this to a module instead of script (allows import/export)
+export {}
+// const getBySel = (selector, ...args) => {
+//   return cy.get(`[data-test=${selector}]`, ...args);
+// };
+// // @ts-ignore
+// Cypress.Commands.add("getBySel", getBySel);
+//
+// //fill basic info
+// const fillBasic = applicant => {
```
Comment (samiran-bs) [inline cypress/support/commands.ts]: remove all commented lines if thee file is needed.

File: package.json [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+    "mocha": "^8.3.2",
+    "mochawesome": "^6.2.2",
+    "ts-loader": "^8.0.7",
+    "typescript": "4.1.2",
+    "webpack": "4.41.2"
+  },
+  "dependencies": {
+    "browserstack-cypress-cli": "^1.8.0"
+  }
+}
```
Comment (samiran-bs) [inline package.json:45]: there is an empty file test2.js below. remove it.

File: bstack-parallel.json [hunk 25]
```diff
@@ -0,0 +1,62 @@
+{
```
Comment (samiran-bs) [inline bstack-parallel.json]: Just a question, do you think placing these cypress*.json and  browserstack*.json in a folder within the root directory, let's say run_conf give more clarity to user. since the number of files have increased.

finally, this if done will have a small impact on the commands.

File: cypress/integration/e2e/test8.spec.ts [hunk 26]
```diff
@@ -0,0 +1,89 @@
+import Login from '../../pageObjects/login'
+import Purchase from '../../pageObjects/purchase'
+import Signin from '../../pageObjects/signin'
+import Address from '../../pageObjects/address'
+import Orders from '../../pageObjects/orders'
+
+
+
+describe('End to End Scenario', function()  {
```
Comment (samiran-bs) [inline cypress/integration/e2e/test8.spec.ts]: when you check the new README.md file, you will realise the need to have more clear names for the test spec files.

There is a section like this. Note: The names of the scenarios need not be exact same - since these are file names in your case but you would definitely need to come up with better names for the files.

## About the tests in this repository

  This repository contains the following Selenium tests:

  - End to End Scenario
  - Login as username
  - Login as Locked User
  - Offers for mumbai geo-location
  - Apply Apple Vendor Filter
  - Apply Lowest to Highest Order By
  - Login as User with no image loaded
  - Login as User with existing Orders

  Note: "End to End Scenario" is the default test in all the single test run profiles. This test also demonstrates the [Page Object Model design pattern](https://www.browserstack.com/guide/page-object-model-in-selenium).

File: bstack-parallel.json (hunk 25 above)
Comment (arvind1998) [reply bstack-parallel.json]: yup...that would make it more compact...will do that

File: cypress/integration/product/test1.spec.ts (hunk 19 above)
Comment (arvind1998) [reply cypress/integration/product/test1.spec.ts]: very :)
